src/Common/src/TypeSystem/IL/ILImporter.cs

```diff
@@ -265,6 +271,12 @@ private void FindJumpTargets()
                         break;
                     case ILOpcode.switch_:
                         {
+                            if (_currentOffset + 3 >= _ilBytes.Length)
```

Comment:
Should this checks rather be in the ReadILUInt32, SkipIL, etc. methods so that they kick in everywhere? E.g. the method can end in the middle of one the labels that follows switch.

Reply:
Yes, I also thought about this. That's what I meant by:

> I am not sure if it would make sense to always verify that we read inside the range of the ilBytes array inside the IL stream reading operations. Do you have any thoughts on this, @jkotas ?

I am just not sure how to handle the return value in an invalid case, or should we simply throw a custom exception and surround the switch with a try, catching that specific exception?

Reply:
I think the situations where a method ends in the middle of an instruction can just abort the verification right away. Throw exception and let it go all the way to the top.

Reply:
I saw other projects using `ThrowHelper` to throw `InvalidProgramExceptions`. Are there any guidelines for throwing an exception for this case?

Reply:
The common ILImporter should just call a method. The specializations of the ILImporter can then throw the kind of exception they wish: ILVerify can throw one with detailed diagnostic info, others should just throw generic InvalidProgram.

Reply:
Ah, I think I understand what you mean now, thanks!